src/components/Modals/NewMessageModal.jsx

```diff
@@ -0,0 +1,215 @@
+        setSuccess(`Message successfully sent to ${message.recipientPodUrl}`);
+        setSuccessTimeout(true);
+        setTimeout(() => {
+          setSuccessTimeout(false);
+        }, 10000);
+      } catch (err) {
+        // TODO: Make sure invalid username is the only possible error
+        setError(err.message);
+      } finally {
+        setOriginalMessage('');
+      }
+    }
+
+    // Re-sorts messages when new message is added to outboxList
+    const outboxMessages = await getMessageTTL(session, 'Outbox', outboxList, podUrl);
+    const sortedOutbox = outboxMessages;
+    sortedOutbox.sort((a, b) => b.uploadDate - a.uploadDate);
+    setOutboxList(sortedOutbox);
+  };
+
+  /* eslint-disable jsx-a11y/label-has-associated-control */
+  return (
+    <Dialog
+      open={showModal}
+      aria-labelledby="new-message-modal"
+      onClose={() => setShowModal(false)}
+    >
+      <Box
+        noValidate
+        sx={{
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'center',
+          alignItems: 'center',
+          padding: '20px',
+          minWidth: '50%',
+          minHeight: '90%'
+        }}
+      >
+        <form onSubmit={(e) => handleSubmit(e)} autoComplete="off">
+          <Typography display="flex" justifyContent="center" variant="h5">
+            {oldMessage ? 'Reply To' : 'New Message'}
+          </Typography>
+          <TextField
+            margin="normal"
+            value={message.recipientPodUrl}
+            type="text"
+            name="recipientPodUrl"
+            id="recipientPodUrl"
+            onChange={(e) => handleChange(e)}
+            required
+            autoFocus
+            label="To"
+            fullWidth
+          />
+          <TextField
+            margin="normal"
+            value={message.title}
+            type="text"
+            name="title"
+            id="title"
+            onChange={(e) => handleChange(e)}
+            required
+            label="Subject"
+            inputProps={{ maxLength: '24' }}
+            fullWidth
+          />
+          <TextField
+            margin="normal"
+            value={oldMessage && originalMessage}
+            type="previousMessage"
+            name="previousMessage"
+            id="previousMessage"
+            label="Previous Message"
+            // helperText="Previous message"
+            // variant="filled"
+            multiline
+            rows={3}
+            InputProps={{
+              readOnly: true
+            }}
+            InputLabelProps={{
+              shrink: true
+            }}
+            fullWidth
+          />
+
+          <TextField
+            margin="normal"
+            value={message.message}
+            type="text"
+            name="message"
+            id="message"
+            onChange={(e) => handleChange(e)}
+            multiline
+            rows={6}
+            label="Enter Message"
+            placeholder="Add a description here"
```

Comment:
Not sure if "description" is accurate for this `placeholder` - or if placeholder is needed at all. None of the other inputs have it, seem to use the label as a direction/placeholder basically.